src/field.rs

```diff
@@ -91,6 +91,52 @@ pub trait Field:
         let int: Self::Integer = (*self).into();
         int.into()
     }
+
+    /// Generic implementation to serialize fields into a buffer. Callers need to make sure
+    /// there is enough capacity to store the value of this field.
+    /// It is less efficient because it operates with generic representation of fields as 16 byte
+    /// integers, so consider overriding it for actual field implementations
+    ///
+    /// ## Errors
+    /// Returns an error if buffer did not have enough capacity to store this field value
+    fn serialize<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        let bytes = Self::Integer::BITS as usize / 8;
+        let raw_value = &self.as_u128().to_le_bytes()[..bytes];
```

Comment:
This is the allocation. Probably harmless and can be optimised out automatically

Reply:
oh wow, I didn't know that. I tried to find it but failed, can you hint me where it happens?

Reply:
Maybe it doesn't heap allocate, but `to_le_bytes()` at least makes a copy of the `u128` as `[u8; 16]`.  It could be aliased to the original value (though not portably) without a copy, but that's not how they did it.

Reply:
not that it matters at this point, but I thought it should be possible to just reinterpret u128 as `[u8; 16]` in LE order? surely for BE we need to make a copy, but I was hopeful that it is not happening here...

Reply:
Let's not compromise portability that way.